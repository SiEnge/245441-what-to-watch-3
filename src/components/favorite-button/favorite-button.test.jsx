import React from "react";
import renderer from "react-test-renderer";
import FavoriteButton from "./favorite-button.jsx";
// import {comment} from "../../utils/test.utils.js";

it(`Render FavoriteButton isFavorite = false`, () => {
  const onFavoriteButtonClick = jest.fn();

  const tree = renderer
    .create(<FavoriteButton
      movieId={1}
      isFavorite={false}
      onClick={onFavoriteButtonClick}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render FavoriteButton isFavorite = true`, () => {
  const onFavoriteButtonClick = jest.fn();

  const tree = renderer
    .create(<FavoriteButton
      movieId={1}
      isFavorite={true}
      onClick={onFavoriteButtonClick}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

