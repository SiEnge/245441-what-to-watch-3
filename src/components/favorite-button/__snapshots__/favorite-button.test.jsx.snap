// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FavoriteButton isFavorite = false 1`] = `
<button
  className="btn btn--list movie-card__button"
  data-favorite={false}
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`Render FavoriteButton isFavorite = true 1`] = `
<button
  className="btn btn--list movie-card__button"
  data-favorite={true}
  onClick={[Function]}
  type="button"
>
  <svg
    height="14"
    viewBox="0 0 18 14"
    width="18"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;
